NDTensors/src/lib/TypeParameterAccessors/src/specify_parameters.jl

```diff
@@ -0,0 +1,48 @@
+using Compat: Compat
+
+Compat.@assume_effects :foldable function specify_parameter(type::Type, pos::Int, param)
+  return _specify_parameter(parameter(type, pos), type, Position(pos), param)
+end
+
+Compat.@assume_effects :foldable function specify_parameter(
+  type::Type, pos::Position, param
+)
+  return _specify_parameter(parameter(type, pos), type, pos, param)
+end
+
+function _specify_parameter(::TypeVar, type::Type, pos::Position, param)
+  return set_parameter(type, pos, TypeParameter(param))
+end
+
+function _specify_parameter(::TypeVar, type::Type, pos::Position, param::Type)
+  return set_parameter(type, pos, param)
+end
+
+@generated function _specify_parameter(
+  ::Union{<:Int,<:DataType}, ::Type{Typ}, pos, param
```

Comment:
An issue with using `Union{<:Int,<:DataType}` here is that type parameters can be any object satisfying `isbitstype`, not just types or `Int`. It seems like you could just leave off the type constraint of the first argument.

Reply:
Yes this makes sense, I used your `is_parameter_specified` function to create a `SpecifiedParameter` struct that has a parameter of `Bool` and rewrote these as multiple dispatch in a trait-like way